### PR TITLE
first try at new wipac ci/cd action

### LINKED ISSUE #49: Publish to PyPI with `wipac-cicd.yml`
See https://github.com/WIPACrepo/wipac-dev-tools/issues/20

`krs` is used as a REST-utility package by clients, including MOU Dash.

### PR DESCRIPTION
**Pull request by dsschult**
Closes #49

### CODE REVIEW

**Comment by dsschult** [comment]
I'm ignoring the user_mgmt tests, as they are removed in #51.  If that PR lands first, it can be incorporated here.  Maybe Monday.

File: requirements.txt [hunk 1]
```diff
@@ -1,14 +1,66 @@
...
-coverage
-requests
-pyjwt[crypto]
-motor
-unidecode
-ldap3
-aio_pika
-flake8
-cachetools
-git+https://github.com/WIPACrepo/rest-tools@v1.1.20#egg=rest_tools
```
Comment (ric-evans) [inline requirements.txt:14]: Some of these may be useful to put into a `requirements-dev.txt` at the root dir.

Comment (ric-evans) [reply requirements.txt:14]: Oh, I see they're in `[options.extras_require].tests`. I'm not sure what's best

Comment (dsschult) [reply requirements.txt:14]: I guess the question is how much we care about `requirements.txt` files, except as a record of what happened in CI.  If we say the authoritative package list is in `setup.cfg` and change the env setup to just doing `pip install -e .[tests]` then the new way works.

Comment (ric-evans) [reply requirements.txt:14]: > I guess the question is how much we care about `requirements.txt` files, except as a record of what happened in CI. If we say the authoritative package list is in `setup.cfg` and change the env setup to just doing `pip install -e .[tests]` then the new way works.

I agree, I think this is the best way to do things (this is also why I've preferred using tox for tests). The only issue is with the mypy action, it installs from `requirements-dev.txt` to lint things like tests.

Comment (dsschult) [reply requirements.txt:14]: I added an issue on the mypy action.  Of course, `krs` doesn't actually do a mypy test right now.   You can tell that I wrote this, as it's not strongly typed. :smile:

**Comment by dsschult** [comment]
Leaving the flake8 errors for now, as those seem to need a separate PR.

**Pull request merged**